ark/phenotyping/cluster.py

```diff
@@ -119,9 +132,13 @@ def train_som(pixel_mat, x_neurons, y_neurons, num_passes,
         # find the winning neuron's coordinates
         winning_coords = winner(pixel_mat.loc[row, :].values, weights)
 
+        # decay both sigma and learning_rate using the decay function
+        decay_sigma = decay_function(sigma, t, num_iters)
+        decay_learning_rate = decay_function(learning_rate, t, num_iters)
```

Comment:
Is this the same as the LR decay in miniSOM? I would assume it only gets decayed after each pass through the dataset

Reply:
Yup, LR needs to be decayed after each iteration in MiniSOM.

Reply:
FlowSOM does it per pass through the dataset

Reply:
@ngreenwald OK, so in a separate PR (which I'll also use to address a potential speed-up idea for training), we should probably update our decay to match that of FlowSOM so we get consistent results.

Reply:
Yes, I think in the long term we'll want the one you have now, but for initial testing we should switch to match FlowSOM. You can add a separate decay function that works across iterations, and one that works across dataset splits. Can you also go through and summarize what the remaining differences are between this and FlowSOM? For example I think you mentioned that the Gaussian function is different for updates?

Reply:
@ngreenwald based on the documentation of `SOM` (contained here: https://www.rdocumentation.org/packages/FlowSOM/versions/1.4.0/topics/SOM), it appears that the biggest difference is in the way the learning rate parameter is being updated. It also appears that R SOM precomputes the neighborhood matrix prior to training (which I guess is a nice speedup) and uses the supremum norm to do so (rather than a Gaussian).

Reply:
Okay great, that can be in a separate PR